syft/source/content_requester.go

```diff
@@ -0,0 +1,48 @@
+package source
+
+import "sync"
+
+type ContentRequester struct {
+	request map[Location][]*FileData
+	lock    sync.Mutex
+}
+
+func NewContentRequester(data ...*FileData) *ContentRequester {
+	requester := &ContentRequester{
+		request: make(map[Location][]*FileData),
+	}
+	for _, d := range data {
+		requester.Add(d)
+	}
+	return requester
+}
+
+func (b *ContentRequester) Add(data *FileData) {
```

Comment:
It might make sense to have this not be a pointer. This would reduce the number of possible states in the system. It also looks like consumers of this function are having to take a reference to data they already have just to be able to call this function.

Reply:
it is important that this remains a pointer since this requester is writing back results to the given `FileData` object upon `Execute()` (this approach will not work otherwise).

Reply:
👍 (We discussed keeping this for now, and keeping an eye out for an approach that can avoid writing back to a given pointer.)